src/Toml/Parser/Value.hs

```diff
@@ -41,14 +59,47 @@ decimalP = zero <|> more
     check :: Maybe Integer -> Parser Integer
     check = maybe (fail "Not an integer") pure
 
+-- | Parser for hexadecimal numbers : included parsing of underscore.
+hexadecimalP :: Parser Integer
+hexadecimalP = zero <|> more
+  where
+    zero, more :: Parser Integer
+    zero = 0 <$ char '_'
+    more = check =<< baseStringToInt 16 . concat <$> sepBy1 (some hexDigitChar) (char '_')
+
+    check :: Maybe Integer -> Parser Integer
+    check = maybe (fail "Not hexadecimal") pure
+
+-- | Parser for octal numbers : included parsing of underscore.
+octalP :: Parser Integer
+octalP = zero <|> more
+  where
+    zero, more :: Parser Integer
+    zero = 0 <$ char '_'
```

Comment:
I think that the `zero` parser is redundant in `octalP`, `binaryP` and `hexadecimalP`. I'm not even sure why it's introduced in `decimalP` 🤔 Would be nice to investigate what tests fail when we remove `zero` from `decimalP`. But `zero` definitely looks redundant in such parsers.

Reply:
Zero in ``decimalP`` is used for leading zeros. Even though I don't understand why number with leading zero is parsed into zero, when official documentation of TOML says ``Leading zeros are not allowed.``. But that could be discussed in a new issue.

Reply:
> I don't understand why number with leading zero is parsed into zero

Yeah, if the current `tomland` implementation allows numbers with leading zeroes, then this problems deserves a separate issue and a solution...